packages/@aws-cdk/aws-appsync/README.md

```diff
@@ -81,6 +81,23 @@ demoDS.createResolver({
 });
 ```
 
+## Imports
+
+Any GraphQL Api can be imported from another stack. Utilizing the `fromXxx`
+functions, you have the ability to add data sources and resolvers through
+a `IGraphQLApi` interface.
+
+```ts
+const importedApi = appsync.GraphQLApi.fromGraphQLApiId(...);
+importedApi.addDynamoDbDataSource(...);
+```
+
+GraphQL Apis can be imported in three ways:
+- `arn` through `GraphQLApi.fromGraphQLApiArn`: `apiId` is generated from parameter
+- `apiId` through `GraphQLApi.fromGraphQLApiId`: `arn` is generated from parameter
+- or both through `GraphQLApi.fromGraphQLApiAttributes`: both are specified
```

Comment:
in this case then i think it might just better to do `attributes` and `ApiId` because as you said it doest get confusing.

Reply:
actually i take that back.. it would be better to just to `attributes` and make the arn optional..

the arn isn't that necessary to be honest in terms of creating resolvers/data sources.. the only thing it would be useful would be for grants